toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/Authenticator.kt

```diff
@@ -63,14 +74,78 @@ public fun Authenticator(
     }
 }
 
+/**
+ * Displays appropriate Authentication UI when an authentication challenge is issued.
+ *
+ * For example, when an [ArcGISAuthenticationChallenge] is issued and the [AuthenticatorState] has a corresponding
+ * [OAuthUserConfiguration], then a Custom Tab will be launched to complete the OAuth sign in.
+ *
+ * This Composable will adapt to the size of its container, allowing flexible layout usage. For alternate behavior,
+ * see [DialogAuthenticator].
+ *
+ * @param authenticatorState the object that holds the state to handle authentication challenges.
+ * @param modifier the [Modifier] to apply to this Authenticator.
+ * @param onBrowserChallengeRequested if not null, this will be called when a authentication challenge is pending
+ * and the browser should be launched. Use this if you wish to handle browser challenges from your own
+ * activity rather than using the [AuthenticationActivity], more information can be found in the
+ * [Authenticator Toolkit README](https://github.com/Esri/arcgis-maps-sdk-kotlin-toolkit/blob/main/toolkit/authentication/README.md).
+ * @see DialogAuthenticator
+ * @since 200.8.0
+ */
+@JvmName("AuthenticatorWithBrowserAuthChallenge")
+@Composable
+public fun Authenticator(
+    authenticatorState: AuthenticatorState,
+    modifier: Modifier = Modifier,
+    onBrowserChallengeRequested: ((BrowserAuthChallenge) -> Unit)
```

Comment:
I think we should keep the use of "pending", i.e. I suggest:
```suggestion
    onPendingBrowserAuthenticationChallenge: ((BrowserAuthenticationChallenge) -> Unit)
```